pkg/config/config.go

```diff
@@ -147,8 +147,8 @@ func checkFilePerms(files ...string) error {
 			}
 		} else {
 			// Assume unix based system (MacOS and Linux)
-			if fInfo.Mode() != 0640 {
-				return errors.E(op, f+" should have 0640 as permission")
+			if fInfo.Mode()&0137 != 0 {
```

Comment:
It's more readable to check `fInfo.Mode() & 077 != 0`. We don't care about the user file permissions, but we want to make sure no one else can access it.

Reply:
Also, `os.Lstat(...)` should probably not be used here since it will return the permissions of a sym link and not the actual file.

Reply:
Yes, it should `os.Stat` Thanks for pointing that out.

077 won't work always. For a file with `-rw-r-----`
```
fInfo.Mode()&0077 -rw-r----- ----r-----
fInfo.Mode()&0137 -rw-r----- ----------
```

On the other hand 0137 is consistent with umask. I will add a note to make it clear how we got there. umask is standard on both macOS and linux

Reply:
0640 was decided earlier, let us keep it and allow scoped permissions under them. I think earlier it was set to 0600 as that is was tools like ssh keys are stored, but there was another user who preferred 0640. 

Personally, I think the files should not be executable at all.

Reply:
If athens creates the config file it should really be created as 600 imho. It might contain secrets as credentials for 3rd party services and such. So maybe change to 600 at creation but allow the file to be readable by group (as decided)?

Reply:
And it should be consistent with the Windows check, wouldn't make sense to threat different platforms in different ways.

Reply:
yes! So, let us go with 0600. I am with you on this. I like it better than 0640 even when it is harder.  Yep, let me make that change as well.

For windows, the only mask that is required is the user bit afaik. I can't really test it as I do not have a windows machine.

Reply:
@leitzler I am confused by this https://github.com/gomods/athens/pull/966#discussion_r239798078. 

Athens does not create any files, we allow users to add files like 
filter.conf and auth files like .netrc and so on. I can do either 0600 / 0640, I have no issues in either. I like 0600 better as it makes sure that auth files (like personal netrc tokens) will not be leaked out to a user in a different group.